great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -2245,6 +2243,87 @@ def _add_expectation_suite(
         self.expectations_store.set(key, expectation_suite, **kwargs)
         return expectation_suite
 
+    def update_expectation_suite(
+        self,
+        expectation_suite: ExpectationSuite,
+    ) -> None:
+        """Update an ExpectationSuite that already exists.
+
+        Args:
+            expectation_suite: The suite to use to update.
+
+        Raises:
+            DataContextError: A suite with the given name does not already exist.
+        """
+        expectation_suite_name = expectation_suite.expectation_suite_name
+        key = ExpectationSuiteIdentifier(expectation_suite_name=expectation_suite_name)
+        if not self.expectations_store.has_key(key):  # noqa: W601
+            raise gx_exceptions.DataContextError(
+                f"expectation_suite with name {expectation_suite_name} does not exist."
+            )
+
+        meta: dict = expectation_suite.meta
+        self._add_expectation_suite(
+            expectation_suite_name=expectation_suite_name,
+            id=expectation_suite.ge_cloud_id,
+            expectations=expectation_suite.expectations,
+            evaluation_parameters=expectation_suite.evaluation_parameters,
+            data_asset_type=expectation_suite.data_asset_type,
+            execution_engine_type=expectation_suite.execution_engine_type,
+            meta=meta,
+            overwrite_existing=True,
+        )
+
+    def add_or_update_expectation_suite(
```

Comment:
Assuming this was a requirement, but why do we need `add`, `update`, and `add_or_update`?

Reply:
Isn't this equivalent to `add` with `overwrite_existing=True`?

Reply:
yup but the purpose of this epic is to standardize CRUD methods across different objects - we'll have add, update, add_or_update, get, and delete for all primary object types.

We want to get away from flags like `overwrite_existing` since it is inconsistently applied across GX. `add` should never overwrite and `update` should fail if the target object doesn't exist. The new `add_or_update` provides a mechanism to bridge the gap between the two.